StripeCore/StripeCore/Source/Helpers/URLEncoder.swift

```diff
@@ -37,7 +37,20 @@ import Foundation
 
     @objc(queryStringFromParameters:)
     public class func queryString(from parameters: [String: Any]) -> String {
-        return query(parameters)
+        return query(parameters, encoder: escape)
+    }
+
+    // For apps linked on or after iOS 17 and aligned OS versions, `URL` automatically percent- and IDNA-encodes invalid
+    // characters to help create a valid URL. See https://developer.apple.com/documentation/foundation/url/3126806-init
+    @objc(queryStringForURLFromParameters:)
+    public class func queryStringForURL(from params: [String: Any]) -> String {
+        let encoder = if #available(iOS 17.0, *) {
```

Comment:
This only applies for apps *linked on* and running on iOS 17, so we might need to check the compiler version as well. But IMO we should rewrite this to skip the escaping and use URLComponents instead, that way we'll have consistent behavior without checking the version.

Reply:
Ah ok, this was this `#if compiler(>=5.9)` part right? 

I think you're right about URLComponents so I'll abandon this hacky fix.

Reply:
I can take on the URLComponents fix, but I'll reuse your test! Thanks for writing this!

Reply:
@davidme-stripe awesome, glad there's something helpful for you here, especially after all your help with #ir-after-deluge